spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java

```diff
@@ -247,9 +249,18 @@ public Object doWithRetry(RetryContext context) throws RuntimeException {
 						public Collection<PartitionInfo> doWithRetry(RetryContext context) throws Exception {
 							Collection<PartitionInfo> partitions = callable.call();
 							// do a sanity check on the partition set
+							int partitionSize = partitions.size();
 							if (partitions.size() < partitionCount) {
-								String topic = partitions.isEmpty() ? "unknown" : partitions.iterator().next().topic();
-								unexpectedPartitionCountHandling.handlePartitionCountTooLow(topic, partitions.size(), partitionCount);
+								if (tolerateLowerPartitionsOnBroker) {
+									logger.warn("The number of expected partitions was: " + partitionCount + ", but "
+											+ partitionSize + (partitionSize > 1 ? " have " : " has ") + "been found instead."
+											+ "There will be " + (partitionCount - partitionSize) + "idle consumers");
```

Comment:
Requires `space` between the `(partitionCount - partitionSize) ` and `idle`. Will fix while merging.